opal/mca/hwloc/hwloc201/hwloc/hwloc/topology-linux.c

```diff
@@ -3419,6 +3422,86 @@ look_sysfsnode(struct hwloc_topology *topology,
   return 0;
 }
 
+//HACK: prevent get not-allowed cpu
+void set_allowed(int* allowed, int num1, int num2)
+{
+    for(int i = num1; i<= num2; i++)
+    {
+        allowed[i] = 1;
+    }
+}
+
+int read_num(char* begin)
+{
+    if(*begin < '0' || *begin > '9') return atoi(begin+1);
+    return atoi(begin);
+}
+
+
+char* find_next_deli(char* begin)
+{
+    char* comma = strchr(begin, ',');
+    char* dash = strchr(begin, '-');
+    char* ret = dash; 
+    if(comma != NULL)
+    {
+        if(dash == NULL || comma < dash) ret = comma;
+    }
+
+    return ret;
+}
+
+
+
+void parse_cpus(int *allowed, char* buf)
+{
+    char *last_explore, *current_explore;
+
+    current_explore = buf;
+    last_explore = current_explore; //last_explore는 마지막으로 탐색한 숫자 첫 자리에 위치
+
+    while( *(++current_explore) != '\n' && *current_explore != '\0')
+    {
+        current_explore = find_next_deli(current_explore);
+        if( current_explore == NULL) current_explore = buf + strlen(buf) - 1;
```

Comment:
I'm not sure what the purpose of this line is.

if current_explore is NULL, why not break out of the loop?